checker.cpp

```diff
@@ -1,17 +1,91 @@
 #include <assert.h>
+#include <iostream>
+using namespace std;
+// enum Range
+/*bool isBPOk(float bpm);
+bool isSpoOk(float spo2);
+bool isRespOk(float respRate);
+bool isVitalOk(float value, float low, float high);
+*/
+const int bp_limits[] = {70,150};
+const int resp_limits[] = {30,95};
 
-bool vitalsAreOk(float bpm, float spo2, float respRate) {
-  if(bpm < 70 || bpm > 150) {
+class Alert{
+  public:
+  virtual void raiseAnAlert(const char* vital, const char *message) = 0;
+  
+};
+
+class AlertInSMS : public Alert{
+  public:
+  virtual void raiseAnAlert(const char* vital, const char *message) override{
+    cout << "send SMS :" << endl;
+  }
+};
+
+class AlertInSoung : public Alert{
+  public:
+  virtual void raiseAnAlert(const char* vital, const char *message) override{
+    cout << "Make Alert sound :" << endl;
+  }
+};
+class CheckVitals{
+  private:
+    Alert* alert;
+  public:
+  CheckVitals(): alert(0){}
+  void setAlert(Alert* alert_requested){
+    alert = alert_requested;
+  }
+  
+  bool isVitalOk(float value, float low, float high){
+    return (value >= low && value <= high);
+  }
+  bool isBPOk(float bpm){
+    if(isVitalOk(bpm,bp_limits[0],bp_limits[1])){
+      return true;
+    }
+    alert->raiseAnAlert("bpm","BP ");
     return false;
-  } else if(spo2 < 90) {
+  }
+
+  bool isSpoOk(float spo2){
+    if(isVitalOk(spo2,90,100)){
+      return true;
+    }
+    alert->raiseAnAlert("spo2","SPO2");
```

Comment:
What is the purpose of the spo2/bpm/etc being passed twice? The original intent of the message was to have low/normal/high indication... since the treatment would be different in those situations.